### DISCUSSION TITLE
Meeting Minutes for June 27, 2024

### DISCUSSION THREAD

**Discussion by dezidizon** [CVWRSM/cvwrsm#102] [Meeting Minutes]
**Agenda:**
1. Discuss which project to prioritize and begin work on under TO2
2. WRIMS Devop Status and estimated budget for the remaining tasks. 
3. Task initiation and funding process. 

**Open Action Items**
- Prepare level of effort, budget, and schedule to perform work for the following projects:
  - WRIMS 2 DevOps
  - IWFM and IDC DevOps
  - WRIMS 3 Development
- Coordinate meetings with Can Dogrul to discuss workflow and repositories for IWFM and IDC.
- Coordinate meeting with @hxiedwr to go over process for building the WRIMS 2 installer and the manual process for creating patch releases.

**Discussions:**
- The funds under TO 2 are intended for work with WRIMS DevOps and development, with IWFM and IDC DevOps, and future work.
  - $100 k of the funding should be allotted for maintenance for 4 years from today
  - $280k of the funding should be allotted for WRIMS 2, WRIMS3, IDC and IWFM DevOps, and miscellaneous work.
- Portion allotted for Task 2 is dedicated to DWR working directly with Paul Hutton.
- @hxiedwr is the Task Order Manager and primary POC for work under TO2.
  - @inazrul should be included in discussions and can help in making decisions.
  - Can Dogrul has the same level of authority to assign work for the IWFM and IDC as @inazrul and @hxiedwr.
- Priority for DWR is producing a WRIMS Package that reproduces what @hxiedwr is producing on his computer. This is the goal in completing the WRIMS 2 DevOps work. After this is done, work is immediately starting with WRIMS 3 development.
  - @hxiedwr has prepared the technical scope for WRIMS 3 and will be sharing the plan with RMA shortly.
- DevOps implementation for IWFM and IDC may be done in parallel with the WRIMS 3 development.
  - There are multiple repositories for IWFM and IDC on SVN. 
  - RMA will meet with Can to understand the repositories and his workflow with IWFM and IDC.
  - When migrating these repositores from SVN to GitHub, the commit history must also be included.  
- @dwr-psandhu discussed including the JCBC in the DevOps effort.
  - Currently, JCBC is committed locally but does not yet have a repository to commit to.
  - Ryan Ripken worked on the JCBC solver.
  - This work may be done in between finishing the WRIMS 2 DevOps and WRIMS 3 development. 
- CalLite maintenance work will involve removing portions of the code related to maps in CalLite.
- Miscellaneous tasks like the solver and ANN, which do not yet have concrete scopes, will be done in an on-demand basis.
- EPPT work is on pause as DWR is internally working on different visualization tools.
  - A senior engineer in their section will take lead at some point.
- For the Excel DSS Plugin work, RMA will produce an installation package after HDR has implemented the updates for HeclLib 7 to the Excel DSS Plugin.
- Work for the VIC Model is currently being done with HDR.
  - DWR prefers RMA's focus to be on WRIMS 2 DevOps and WRIMS 3 development. 
- @rma-psmorris discussed that there is room for parallel development.
  - RMA will decide on which projects to work on in parallel
- Level of effort, budgets, and schedules need to be prepared for the following projects to ensure that there is enough funding and that work proceeds.
  -  Completing WRIMS 2 DevOps
  -  IWFM and IDC DevOps
  - WRIMS 3 Development

**Comment by inazrul** [comment]
I would like to know the status of the WRIMS Devops current status and budget estimate on the remaining tasks.

**Comment by adamkorynta** [comment]
Referenced during the meeting: transition from SVN to GitHub
- https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/importing-a-subversion-repository
- https://git-scm.com/docs/git-svn
